pandas/tests/test_base.py

```diff
@@ -972,6 +972,18 @@ def test_memory_usage(self):
             diff = res_deep - sys.getsizeof(o)
             self.assertTrue(abs(diff) < 100)
 
+    def test_searchsorted(self):
+        # See gh-12238
+        for o in self.objs:
+            # sanity check
+            index = np.searchsorted(o, max(o))
+            self.assertTrue(0 <= index <= len(o))
+
+            # check that the 'sorter' argument can be
+            # passed in via numpy's searchsorted call
+            index = np.searchsorted(o, max(o), sorter=range(len(o)))
```

Comment:
Not sure if I do in the context of compatibility enforcement.  This test first tests that behaviour is well-defined when you "correctly" call `np.searchsorted` (i.e. with only the arguments `pandas` wants) and then "incorrectly" (i.e. with all the arguments that `numpy` would accept), so I think I've covered all the necessary `numpy` compat cases IMO.